DABI.js

```diff
@@ -0,0 +1,677 @@
+        var a = d.evaluateThing(d.detect, e, b);
+        if (a.length > 0 && a[0]) {
+            return f
+        }
+    }
+    return undefined
+};
+FW.getScraper = function(b, a) {
+    var c = FW.detectWeb(b, a);
+    return FW._scrapers.filter(function(d) {
+        return (d.evaluateThing(d.itemType, b, a) == c) && (d.evaluateThing(d.detect, b, a))
+    })[0]
+};
+FW.doWeb = function(c, a) {
+    var b = FW.getScraper(c, a);
+    b.makeItems(c, a, [], function(f, e, g, d) {
+        e.callHook("scraperDone", f, g, d);
+        if (!f.title) {
+            f.title = ""
+        }
+        f.complete()
+    }, function() {
+        Zotero.done()
+    });
+    Zotero.wait()
+};
+
+function detectWeb(doc, url) {
+    //namespace code
+    var namespace = doc.documentElement.namespaceURI;
+    var nsResolver = namespace ? function(prefix) {
+        if (prefix == 'x') return namespace;
+        else return null;
+    } : null;
+
+    if (doc.title.match("DABI. Datensatz Vollanzeige")) {
+        return "journalArticle";
+    } else if (doc.title.match("DABI: Rechercheergebnis")) {
+        var treffer = doc.evaluate('//*[text()="Keine Treffer für die Suche nach:"]', doc, nsResolver, XPathResult.ANY_TYPE, null).iterateNext();
+        if (treffer == null) {
+            return "multiple";
+        }
+    }
+}
+
+function doWeb(doc, url) {
+    //namespace code
+    var namespace = doc.documentElement.namespaceURI;
+    var nsResolver = namespace ? function(prefix) {
+        if (prefix == 'x') return namespace;
+        else return null;
+    } : null;
+
+    //variable declarations
+    var articles = new Array();
+    var items = new Object();
+    var nextLink;
+
+    //If Statement checks if page is a Search Result, then saves requested Items
+    if (detectWeb(doc, url) == "multiple") {
+        var links = doc.evaluate('//tr/td[1]/a', doc, nsResolver, XPathResult.ANY_TYPE, null);
+        var title = doc.evaluate('//tr/td[3]', doc, nsResolver, XPathResult.ANY_TYPE, null);
+
+        while (nextLink = links.iterateNext()) {
+            items[nextLink.href] = title.iterateNext().textContent;
+
+        }
+        items = Zotero.selectItems(items);
+        for (var i in items) {
+            articles.push(i);
+        }
+
+    } else if (detectWeb(doc, url) == "journalArticle") {
+        //saves single page items
+        articles = [url];
+    } else {
+        return false;
+    }
+
+    //Tells Zotero to process everything. Calls the "scrape" function to do the dirty work.
+    Zotero.Utilities.processDocuments(articles, scrape, function() {
+        Zotero.done();
+    });
+    Zotero.wait();
+    //Translator is FINISHED after running this line. Note: code doesn't run from top to bottom only.
+}
+
+//The function used to save well formatted data to Zotero
+function associateData(newItem, items, field, zoteroField) {
+    if (items[field]) {
+        newItem[zoteroField] = items[field];
+    }
+}
```

Comment:
Personally, I wouldn't make a function just to avoid an `if` statement, but if you really want to, I would declare this function inside scrape and then avoid passing in `newItem` and `items`, since those would be in scope anyway.